### ISSUE TITLE
Exception: Clamp always fails

### ISSUE THREAD

**Issue by agroce** [vyperlang/vyper#1732]
Same setup as previous TSTL reports.

```
@public
def f0():
    lv0: decimal = MAX_DECIMAL + MAX_DECIMAL
    raise("0")
```

vyper vfile.84642.vy
Error compiling: vfile.84642.vy
Exception: Clamp always fails

**Comment by fubuloubu** [comment]
That's a weird exception, but probably correct behavior since that's an out of bounds value for Decimal

**Comment by agroce** [comment]
Be nice to have line no. etc. and a more informative message, but fair enough.

**Comment by fubuloubu** [comment]
Yeah, the error message is definitely off

**Comment by iamdefinitelyahuman** [comment]
This issue appears to have moved..  the code is now compiling, but fails at runtime from an overflow.

We should be doing compile-time simplifications on `decimal` literals in [`parser/expr.py`](https://github.com/vyperlang/vyper/blob/c9f61620b0b73ad7cf07332f34ce4ce2440cf8fa/vyper/parser/expr.py#L439) so that this doesn't compile.

**Comment by iamdefinitelyahuman** [comment]
Fixed in #2029
